src/app/containers/NewSignupContainer/PaymentStep/PaymentStep.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Alert, Payment, usePayment, PrimaryButton, Field, Label, Row } from 'react-components';
+import { c } from 'ttag';
+import { PAYMENT_METHOD_TYPES, CYCLE, CURRENCIES } from 'proton-shared/lib/constants';
+
+// TODO: loading / check payment?
+const PaymentStep = ({ onPaymentDone, paymentAmount, model, children }) => {
+    const { method, setMethod, parameters, canPay, setParameters, setCardValidity } = usePayment();
+
+    const handlePaymentDone = () => onPaymentDone(model, parameters);
+
+    return (
+        <>
+            <h3>{c('Title').t`Select a payment method`}</h3>
+            <Row>
+                <div>
+                    <Alert>{c('Info').t`Your payment details are protected with TLS encryption and Swiss laws`}</Alert>
+                    <Payment
+                        type="signup"
+                        method={method}
+                        amount={paymentAmount}
+                        cycle={model.cycle}
+                        currency={model.currency}
+                        parameters={parameters}
+                        onParameters={setParameters}
+                        onMethod={setMethod}
+                        onValidCard={setCardValidity}
+                        onPay={handlePaymentDone}
+                    />
+                    {method === PAYMENT_METHOD_TYPES.CARD && (
+                        <Row>
+                            <Label></Label>
```

Comment:
`<Label />` require a `children` no?

Reply:
Yes, haven't figured out the layout here, used label for simplicity, but it just basically has to be a button floated to the right, but the form has awkward width.